sale_order_product_recommendation_secondary_unit/wizards/sale_order_recommendation.py

```diff
@@ -0,0 +1,93 @@
+# Copyright 2019 David Vidal <david.vidal@tecnativa.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+from odoo.addons import decimal_precision as dp
+from odoo.tools.float_utils import float_compare, float_round
+
+
+class SaleOrderRecommendation(models.TransientModel):
+    _inherit = 'sale.order.recommendation'
+
+    @api.model
+    def _prepare_wizard_line(self, vals, order_line=False):
+        res = super()._prepare_wizard_line(vals, order_line)
+        secondary_uom_id = (
+            order_line and order_line.secondary_uom_id or
+            vals.get('product_id') and
+            vals['product_id'].sale_secondary_uom_id)
+        secondary_uom_qty = False
+        if not order_line and secondary_uom_id:
+            secondary_uom_qty = (
+                res.get('units_included', 0) / secondary_uom_id.factor)
```

Comment:
(secondary_uom_id.factor or 1.0)

Reply:
Looks like a constraint should be applied to that field, isn't it? I mean, how the factor could be 0?

Reply:
I mean, in the same way regular uoms do: https://github.com/odoo/odoo/blob/11.0/addons/product/models/product_uom.py#L42-L45

Reply:
But this factor is this (I think)
https://github.com/OCA/product-attribute/blob/11.0/product_secondary_unit/models/product_second_unit.py#L25

Reply:
Changed

Reply:
oks...